custom_components/tuya_local/devices/alecoair_d12_home_dehumidifier.yaml

```diff
@@ -0,0 +1,235 @@
+name: AlecoAir D12 Home dehumidifier
+products:
+  - id: hp1sjssyokokbrsg
+#    name: AlecoAir D12 Home
+primary_entity:
+  entity: humidifier
+  class: dehumidifier
+  dps:
+    - id: 1
+      name: switch
+      type: boolean
+      mapping:
+        - dps_val: false
+          icon: "mdi:power"
+          icon_priority: 1
+        - dps_val: true
+          icon: "mdi:power"
+          icon_priority: 4
+    - id: 2
+      name: mode
+      type: integer
+      mapping:
+        - dps_val: Continuous
+          value: 1
+          icon: mdi:fan
+          icon_priority: 3
+        - dps_val: Auto
+          value: 0
+          icon: "mdi:fan-auto"
+          icon_priority: 3
+    - id: 4
+      name: humidity
+      type: integer
+      range:
+        min: 30
+        max: 80
+      mapping:
+        - step: 5
+    - id: 11
+      name: error
+      type: bitfield
+      # Docs mention E1 and E2, it isn't clear which is tank full, so for now
+      # treat both as that.
+      mapping:
+        - dps_val: 0
+          value: OK
+        - dps_val: 1
+          value: E1
+          icon: "mdi:cup-water"
+          icon_priority: 2
+        - dps_val: 2
+          value: E2
+          icon: "mdi:cup-water"
+          icon_priority: 2
+secondary_entities:
+  - entity: switch
+    name: Inside Drying
+    icon: "mdi:air-humidifier"
+    dps:
+      - id: 101
+        name: switch
+        type: boolean
+        mapping:
+          - dps_val: false
+            icon: "mdi:air-humidifier-off"
+            icon_priority: 1
+          - dps_val: true
+            icon: "mdi:air-humidifier"
+            icon_priority: 4
+  - entity: select
+    name: Countdown
+    icon: "mdi:timer"
+    category: config
+    dps:
+      - id: 12
+        name: option
+        type: integer
+        mapping:
+          - dps_val: Cancel
+            value: 0
+            icon: mdi:cancel
+            icon_priority: 3
+          - dps_val: 1 hour
+            value: 1
+            icon: "mdi:store-clock"
```

Comment:
I think you have misunderstood the icons here.  These icons do not appear in the list of options, they are applied to the entity when the specified value is selected.  I think just the "mdi:timer" icon is enough to convey the information that this entity is for the timer, but "mdi:store-clock" would work too but does not need setting for every value, just replace the "mdi:timer" above.  In particular the "mdi:cancel" icon is ambiguous, so I would leave that out.

Reply:
Based on the info on #328, which is for D16 and D14 models which seem to match this same layout, there may also be a dp id 13 for the remaining time, which I guess needs to be optional, and is maybe only sent by the device when the timer is running.